### DISCUSSION TITLE
What is the order in which scripts are executed?

### DISCUSSION THREAD

**Discussion by leo-amelyan** [grate-devs/grate#462] [Q&A] [answered]
If I have the `up` directory with multiple script files, what is the order in which script files are executed?  Is it alphabetical? I couldn't find anything in documentation.

**Comment by leo-amelyan** [comment]
@erikbra could you point me to documentation about this?

**Reply by erikbra** [reply, marked as answer]
Hi, sorry if this is not clear in the documentation. Please feel free to submit a PR on the documentation if you have any suggestions to how you could get startet more easily.

It is mentioned, albeit only in passing, here: https://erikbra.github.io/grate/getting-started/#how-it-works

And, to make it evident: YES, the order of execution is alphabetical :)

Of course, your requirements may vary, but I usually number the scripts in the one-time folders, like this:

0001_create_schema_X.sql
0002_create_table_A.sql
0003_create_table_B.sql
0004_add_missing_column_Y_to_table_A.sql

etc. You might not have the need to grow above 999 files before you start using other characters, so a pattern of `001_scriptname` might be sufficient. But you know this best yourself :)

The "anytime" scripts (e.g. stored procedures), I don't tend to number, myself, as these can change at any time, and inter-dependencies between them can change. But I think people use different strategies here.
